sha1solver/sha1solver.c

```diff
@@ -0,0 +1,141 @@
+#include "sha1.h"
+#include "stdio.h"
+#include "string.h"
+#include "stdlib.h"
+
+#define hex(c) ((*(c)>='a')?*(c)-'a'+10:(*(c)>='A')?*(c)-'A'+10:*(c)-'0') 
+
+int SHA1Progress(
+    char *hash_out,
+    char *str,
+    int len,
+    int poz_start,
+    int poz_end,
+    SHA1_CTX * ctx,
+    char *pattern,
+    int pattern_len,
+    int r,
```

Comment:
What is `r` and `s`? Can more human-readable names be applied?

Reply:
I have added more comments